core/src/main/java/nl/nn/adapterframework/receivers/Receiver.java

```diff
@@ -726,6 +726,33 @@ public void configure() throws ConfigurationException {
 		}
 	}
 
+	private String generateName() {
+		String name = null;
+		if (getListener()!=null) {
+			if(StringUtils.isNotEmpty(getListener().getName())) {
+				name = getListener().getName() + " " + ClassUtils.nameOf(getListener());
```

Comment:
I would prefer to extend the Receiver name with the listener-classname only when there are multiple unnamed Receivers in the Adapter that have the same name.